Source/Core/DolphinUWP/App.cpp

```diff
@@ -0,0 +1,331 @@
+      openPicker.FileTypeFilter().Append(L".gcm");
+      openPicker.FileTypeFilter().Append(L".gcz");
+
+      auto file = co_await openPicker.PickSingleFileAsync();
+      if (file)
+      {
+        path = winrt::to_string(file.Path().data());
+      }
+    }
+
+    CoreWindow window = CoreWindow::GetForCurrentThread();
+    void* abi = winrt::get_abi(window);
+
+    WindowSystemInfo wsi;
+    wsi.type = WindowSystemType::UWP;
+    wsi.render_window = abi;
+    wsi.render_surface = abi;
+    wsi.render_width = window.Bounds().Width;
+    wsi.render_height = window.Bounds().Height;
+
+    auto navigation = winrt::Windows::UI::Core::SystemNavigationManager::GetForCurrentView();
+
+    // UWP on Xbox One triggers a back request whenever the B button is pressed
+    // which can result in the app being suspended if unhandled
+    navigation.BackRequested([](const winrt::Windows::Foundation::IInspectable&,
+                                const BackRequestedEventArgs& args) { args.Handled(true); });
+
+    GAMING_DEVICE_MODEL_INFORMATION info = {};
+    GetGamingDeviceModelInformation(&info);
+    if (info.vendorId == GAMING_DEVICE_VENDOR_ID_MICROSOFT)
+    {
+      HdmiDisplayInformation hdi = HdmiDisplayInformation::GetForCurrentView();
+      if (hdi)
+      {
+        wsi.render_width = hdi.GetCurrentDisplayMode().ResolutionWidthInRawPixels();
+        wsi.render_height = hdi.GetCurrentDisplayMode().ResolutionHeightInRawPixels();
+      }
+    }
+
+    std::unique_ptr<BootParameters> boot =
+        BootParameters::GenerateFromFile(path, BootSessionData("", DeleteSavestateAfterBoot::No));
+
+    UICommon::SetUserDirectory(winrt::to_string(ApplicationData::Current().LocalFolder().Path()));
+    UICommon::CreateDirectories();
+    UICommon::Init();
+    UICommon::InitControllers(wsi);
+
+    if (!BootManager::BootCore(std::move(boot), wsi))
+    {
+      fprintf(stderr, "Could not boot the specified file\n");
+    }
+  }
+
+  void SetWindow(CoreWindow const& w) { w.Closed({this, &App::OnClosed}); }
+
+  void OnClosed(const IInspectable&, const winrt::Windows::UI::Core::CoreWindowEventArgs& args)
+  {
+    m_shutdown_requested.Set();
+  }
+
+  void OnActivate(const winrt::Windows::ApplicationModel::Core::CoreApplicationView&,
+                  const winrt::Windows::ApplicationModel::Activation::IActivatedEventArgs& args)
+  {
+    std::stringstream filePath;
+
+    if (args.Kind() == Windows::ApplicationModel::Activation::ActivationKind::Protocol)
+    {
+      auto protocolActivatedEventArgs{
+          args.as<Windows::ApplicationModel::Activation::ProtocolActivatedEventArgs>()};
+      auto query = protocolActivatedEventArgs.Uri().QueryParsed();
+
+      for (uint32_t i = 0; i < query.Size(); i++)
+      {
+        auto arg = query.GetAt(i);
+
+        // parse command line string
+        if (arg.Name() == winrt::hstring(L"cmd"))
+        {
+          std::string argVal = winrt::to_string(arg.Value());
```

Comment:
Will this handle encodings correctly? (If the system codepage isn't UTF-8, I don't think it will)

Reply:
This was tested with RetroPass which makes use of the protocol/args system. I will have to check if it's possible the encoding will be different, but seeing as this all has to go through the WinRT/Core App stuff, I don't think it will receive any other encoding.